src/traces/scattergl/index.js

```diff
@@ -13,10 +13,12 @@ var ScatterGl = {};
 ScatterGl.attributes = require('./attributes');
 ScatterGl.supplyDefaults = require('./defaults');
 ScatterGl.colorbar = require('../scatter/colorbar');
+ScatterGl.hoverPoints = require('../scatter/hover');
 
 // reuse the Scatter3D 'dummy' calc step so that legends know what to do
-ScatterGl.calc = require('../scatter3d/calc');
+ScatterGl.calc = require('../scatter/calc');
```

Comment:
> we can effortlessly get the mouseContainer back, it is just a matter of 2 commented LOC.

Ok, but the mouseContainer sits on top of the cartesian drag layers, so how will the selection handler gets its data from?

> I will try to optimize snap-points-2d in gl-scatter2d update,

That's cool, but the real bottleneck is in `scatter/calc.js` when comparing this branch to `master`.